components/browser/state/src/main/java/mozilla/components/browser/state/reducer/ContentStateReducer.kt

```diff
@@ -217,8 +219,79 @@ internal object ContentStateReducer {
             is ContentAction.UpdateDesktopModeAction -> updateContentState(state, action.sessionId) {
                 it.copy(desktopMode = action.enabled)
             }
-            is ContentAction.UpdatePermissionHighlightsStateAction -> updateContentState(state, action.sessionId) {
-                it.copy(permissionHighlights = action.highlights)
+            is UpdatePermissionHighlightsStateAction.NotificationChangedAction -> updateContentState(
```

Comment:
nit: Does it make sense to create a `update***State()` helper (which uses `updateContentState()`) to avoid the `it.copy(permissionHighlights = it.permissionHighlights.copy( .... ))` wrapper on every line?

Reply:
Let me give it a try 👍🏽